packages/components/src/components/Heading/Heading.tsx

```diff
@@ -0,0 +1,44 @@
+import { TextProps, TypographyComponent } from "../Text/Text";
+import React from "react";
+import { styled } from "twin.macro";
+
+type HeadingElement = "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
+
+export type HeadingProps = TextProps & {
+  /**
+   * This prop can be used to specify which size heading should
+   * actually be rendered, in the case that you want to render an element
+   * as one heading but style it as if it were another. If both an `as` prop
+   * and a `size` prop are passed, the `as` will be used to determine the element
+   * and the `size` will be used to determine the styling.
+   */
+  as?: HeadingElement;
+};
+
+const HeadingComponent = styled(TypographyComponent).attrs({
+  bold: true,
```

Comment:
Currently in traject, all headings are indeed bold. so I'm comfortable enforcing this, and we can revisit if there's a design need later

also this is how the component is rendered in DOM:
<img width="600" alt="Screen Shot 2021-02-05 at 12 15 11 PM" src="https://user-images.githubusercontent.com/15840841/107070467-67022c00-67b1-11eb-9cb7-7ff29d114ce5.png">